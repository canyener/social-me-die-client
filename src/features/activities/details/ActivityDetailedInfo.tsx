import { observer } from 'mobx-react-lite'
import React from 'react'
import { Segment, Grid, Icon } from 'semantic-ui-react'

import { IActivity } from '../../../app/models/activity'

const ActivityDetailedInfo: React.FC<{ activity: IActivity }> = ({ activity }) => {
  return (
    <Segment.Group>
      <Segment attached='top'>
        <Grid>
          <Grid.Column width={1}>
            <Icon size='large' color='teal' name='info' />
          </Grid.Column>
          <Grid.Column width={15}>
            <p>{activity.description}</p>
          </Grid.Column>
        </Grid>
      </Segment>
      <Segment attached>
        <Grid verticalAlign='middle'>
          <Grid.Column width={1}>
            <Icon name='calendar' size='large' color='teal' />
          </Grid.Column>
          <Grid.Column width={15}>
            <span>{activity.date!.toLocaleDateString('tr-TR')}</span>
          </Grid.Column>
        </Grid>
      </Segment>
      <Segment attached>
        <Grid verticalAlign='middle'>
          <Grid.Column width={1}>
            <Icon name='marker' size='large' color='teal' />
          </Grid.Column>
          <Grid.Column width={11}>
            <span>
              {activity.venue}, {activity.city} 
            </span>
          </Grid.Column>
        </Grid>
      </Segment>
    </Segment.Group>
  )
}

export default observer(ActivityDetailedInfo)