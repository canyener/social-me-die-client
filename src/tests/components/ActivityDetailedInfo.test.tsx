import { shallow } from 'enzyme'

import ActivityDetailedInfo from '../../features/activities/details/ActivityDetailedInfo'

import { fakeActivities } from '../fixtures/seed'

test('Should render ActivityDetailedInfo, correctly', () => {
  const wrapper = shallow(<ActivityDetailedInfo activity={fakeActivities[0]} />)

  expect(wrapper).toMatchSnapshot()
})