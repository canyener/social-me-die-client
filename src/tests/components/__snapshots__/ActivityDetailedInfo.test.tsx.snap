// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ActivityDetailedInfo, correctly 1`] = `
<SegmentGroup>
  <Segment
    attached="top"
  >
    <Grid>
      <GridColumn
        width={1}
      >
        <Icon
          as="i"
          color="teal"
          name="info"
          size="large"
        />
      </GridColumn>
      <GridColumn
        width={15}
      >
        <p>
          Fake description
        </p>
      </GridColumn>
    </Grid>
  </Segment>
  <Segment
    attached={true}
  >
    <Grid
      verticalAlign="middle"
    >
      <GridColumn
        width={1}
      >
        <Icon
          as="i"
          color="teal"
          name="calendar"
          size="large"
        />
      </GridColumn>
      <GridColumn
        width={15}
      >
        <span>
          2020-02-02T16:16:16
        </span>
      </GridColumn>
    </Grid>
  </Segment>
  <Segment
    attached={true}
  >
    <Grid
      verticalAlign="middle"
    >
      <GridColumn
        width={1}
      >
        <Icon
          as="i"
          color="teal"
          name="marker"
          size="large"
        />
      </GridColumn>
      <GridColumn
        width={11}
      >
        <span>
          Fake Venue
          , 
          Fake City
        </span>
      </GridColumn>
    </Grid>
  </Segment>
</SegmentGroup>
`;
